19 - Dados los valores de x: 0.2, 0.4, 0.6, 0.8, 1.0, 1.2, 1.4, 1.6 y la función:

$$ 𝑓(𝑥) = 8 𝑠𝑒𝑛^2 (\frac{𝜋𝑥}{12} + 𝑒^{−𝑥} − 0.25) $$

Determine el valor aproximado de 𝑓(1.48). Además, calcule el valor exacto y el error de aproximación. Use 9 decimales

In [15]:
from math import exp, pi, sin

from metodos_interpolacion import trazador_cubico_natural

def f_de_x(x):
    return 8 * (sin((((pi * x) /12)) + exp(x) - 0.25))**2


datos_x = [0.2, 0.4, 0.6, 0.8, 1.0, 1.2, 1.4, 1.6]

resultado = trazador_cubico_natural(datos_x, [f_de_x(x) for x in datos_x])

contador = 1
for pol in resultado[2]:
    print(f"El trazador {contador} es:", end="")
    display(pol)
    contador += 1

print(
    "Dado que nos piden el valor aproximado en la función "
    "para f(1.48) utilizamos el trazador que corresponde al "
    "intervalo entre 1.4 y 1.6, es decir el trazador 7 \n"
    )

def trazador_7(x):
    return (
        7.46352328945114 * x + 110.036676510311 * (0.714285714285714 * x - 1)**3
        - 47.1585756472761 * (0.714285714285714 * x - 1)**2 
        - 4.56834722009506
    )

valor_aproximado = round(trazador_7(1.48), 9)
print("El valor aproximado de f(1.48) evaluado en el trazador 7 es: ", valor_aproximado, "\n")
valor_exacto = round(f_de_x(1.48), 9)
print("El valor exacto es de: ", valor_exacto, "\n")
error = abs(valor_exacto - valor_aproximado)
print("El error es: ", error, "\n")

El trazador 1 es:

9.11262470472819*x - 6.7048119026029*(x - 0.2)**3 + 4.0129762620652

El trazador 2 es:

2.01872460967822*x - 5.78420715409563*(x - 0.4)**3 - 4.02288714156174*(x - 0.4)**2 + 6.79689780486436

El trazador 3 es:

-10.9021285377913*x + 2.78994823388703*(x - 0.6)**3 - 7.49341143401912*(x - 0.6)**2 + 14.3422205504509

El trazador 4 es:

-19.9858593054379*x + 20.8521232825976*(x - 0.8)**3 - 5.8194424936869*(x - 0.8)**2 + 21.3317882930784

El trazador 5 es:

-6.83078444924459*x + 35.0625401473583*(x - 1.0)**3 + 6.69183147587163*(x - 1.0)**2 + 8.11075272339845

El trazador 6 es:

24.9997878797992*x - 149.154777629306*(0.833333333333333*x - 1)**3 + 39.9302720125728*(0.833333333333333*x - 1)**2 - 29.5377604912404

El trazador 7 es:

7.46352328945114*x + 110.036676510311*(0.714285714285714*x - 1)**3 - 47.1585756472761*(0.714285714285714*x - 1)**2 - 4.56834722009506

Dado que nos piden el valor aproximado en la función para f(1.48) utilizamos el trazador que corresponde al intervalo entre 1.4 y 1.6, es decir el trazador 7 

El valor aproximado de f(1.48) evaluado en el trazador 7 es:  6.344211687 

El valor exacto es de:  7.737977347 

El error es:  1.3937656600000006 



Dado que nos piden el valor aproximado en la función para f(1.48) utilizamos el trazador que corresponde al intervalo entre 1.4 y 1.6, es decir el trazador 7, el cual es:

$$ 7.46352328945114*x + 110.036676510311*(0.714285714285714*x - 1)^3 - $$ 
$$ 47.1585756472761*(0.714285714285714*x - 1)^2 - 4.56834722009506 $$

si evaluamos f(1.48) en este trazador el resultado es: 6.344211687

El valor aproximado es:  6.344211687

El valor exacto es de:  7.737977347

El error es:  1.3937656600000006